request.js

```diff
@@ -866,30 +877,50 @@ Request.prototype.onRequestResponse = function (response) {
   var self = this
 
   if (self.timing) {
-    self.timings.response = now()
+    self.timings.response = now() - self.startTimeNow
   }
 
   debug('onRequestResponse', self.uri.href, response.statusCode, response.headers)
   response.on('end', function() {
     if (self.timing) {
-      self.timings.end = now()
+      self.timings.end = now() - self.startTimeNow
+      response.timingStart = self.startTime
 
-      self.timings.dns = self.timings.socket - self.timings.start
-      self.timings.tcp = self.timings.connect - self.timings.socket
-      self.timings.firstByte = self.timings.response - self.timings.connect
-      self.timings.download = self.timings.end - self.timings.response
-      self.timings.total = self.timings.end - self.timings.start
+      // fill in the blanks for any periods that didn't trigger, such as
+      // no lookup or connect due to keep alive
+      if (!self.timings.socket) {
+        self.timings.socket = 0
+      }
+      if (!self.timings.lookup) {
+        self.timings.lookup = self.timings.socket
+      }
+      if (!self.timings.connect) {
+        self.timings.connect = self.timings.lookup
+      }
+      if (!self.timings.response) {
```

Comment:
I'm not sure it's even possible, but I added it here for consistency and to protect against edge cases.